src/err/mod.rs

```diff
@@ -494,7 +506,7 @@ impl PyErr {
             // pointer.
             let traceback = self.traceback_bound(py);
             ffi::PyErr_Display(
-                self.get_type(py).as_ptr(),
+                self.get_type_bound(py).as_ptr(),
```

Comment:
This `.as_ptr()` now creates a dangling pointer; maybe want to review my migration note in #3805 and see if the wording there makes sense? Probably I should add an example like this there.

To avoid we need to store the `bound` in a variable like we do for `traceback`.